### PR TITLE
Perform method used DB connection as it was "user supplied" and did not close it

### PR DESCRIPTION
**Pull request by cstamas**
https://issues.sonatype.org/browse/NEXUS-8763

CI
http://bamboo.s/browse/NX3-OSSF394

### CODE REVIEW

**Comment by mrprescott** [comment]
+1

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageFacetImpl.java [hunk 1]
```diff
@@ -226,16 +226,20 @@ public void unregisterHookSupplier(final Supplier<StorageTxHook> hookSupplier) {
 
   @Override
   public <T> T perform(final Operation<T> operation) {
-    return perform(databaseInstanceProvider.get().acquire(), operation);
```
Comment (mcculls) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageFacetImpl.java]: If you change this to:
```
try (ODatabaseDocumentTx db = databaseInstanceProvider.get().acquire()) {
  return perform(db, operation);
}
```
and drop the other changes then you get my vote ;)

**Comment by mcculls** [comment]
+1

**Comment by kellyrob99** [comment]
+1

**Comment by cstamas** [comment]
100f62ddd6595498d26a0064e92ec081aff5edbc

**Pull request closed without merging**